src/POData/Providers/Metadata/SimpleMetadataProvider.php

```diff
@@ -289,15 +289,19 @@ public function addEntityType(\ReflectionClass $refClass, $name)
     private function createResourceType(
         \ReflectionClass $refClass,
         $name,
-        $typeKind
+        $typeKind,
+        $isAbstract = false,
+        $baseType = null
     ) {
         if (array_key_exists($name, $this->resourceTypes)) {
             throw new InvalidOperationException('Type with same name already added');
         }
-
+        if(null !== $baseType){
+            $baseType = $this->oDataEntityMap[$baseType->getFullName()];
```

Comment:
Scrutinizer is complaining about assigning to a non-ref parameter - rename this sucker ?